common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordReadMarshallable.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.common.chronicle;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import com.ericsson.bss.cassandra.ecaudit.common.record.AuditRecord;
+import com.ericsson.bss.cassandra.ecaudit.common.record.SimpleAuditOperation;
+import com.ericsson.bss.cassandra.ecaudit.common.record.SimpleAuditRecord;
+import com.ericsson.bss.cassandra.ecaudit.common.record.Status;
+import net.openhft.chronicle.core.io.IORuntimeException;
+import net.openhft.chronicle.wire.ReadMarshallable;
+import net.openhft.chronicle.wire.WireIn;
+import org.jetbrains.annotations.NotNull;
+
+public class AuditRecordReadMarshallable implements ReadMarshallable
+{
+    private AuditRecord auditRecord;
+
+    @Override
+    public void readMarshallable(@NotNull WireIn wire) throws IORuntimeException
+    {
+        if (auditRecord != null)
+        {
+            throw new IORuntimeException("Tried to read from wire with used marshallable");
+        }
+
+        short version = wire.read(WireTags.KEY_VERSION).int16();
+
+        if (version != WireTags.VALUE_VERSION_CURRENT)
+        {
+            throw new IORuntimeException("Unsupported record version: " + version);
+        }
+
+        String type = wire.read(WireTags.KEY_TYPE).text();
+
+        SimpleAuditRecord.Builder builder = SimpleAuditRecord.builder();
+
+        builder.withTimestamp(wire.read(WireTags.KEY_TIMESTAMP).int64());
+
+        try
```

Comment:
Maybe if you break out the value-getting into separate methods (that handles all the try/catch/converting) this method will be shorter and simpler to read...

Then builder.with-methods could be "stacked" on each other...
builder.withX(getX())
    .withY(getY())
    .withZ(getZ())
    ....
    .build()

Reply:
Done